src/coreclr/interpreter/compiler.cpp

```diff
@@ -3460,7 +3494,159 @@ int InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
                         break;
                 }
                 break;
+            case CEE_NEWARR:
+            {
+                CHECK_STACK(1);
+
+                uint32_t token = getU4LittleEndian(m_ip + 1);
+
+                CORINFO_RESOLVED_TOKEN resolvedToken;
+                ResolveToken(token, CORINFO_TOKENKIND_Newarr, &resolvedToken);
+
+                CORINFO_CLASS_HANDLE arrayClsHnd = resolvedToken.hClass;
+                CorInfoHelpFunc helperFunc = m_compHnd->getNewArrHelper(arrayClsHnd);
+                void* pHelper = m_compHnd->getHelperFtn(helperFunc);
```

Comment:
btw, `GetDataItemIndexForHelperFtn` exists now. But it may complicate the implementation of INTOP_NEWARR on the exec side since it supports the indirect path.

Reply:
We can extend GetDataItemIndexForHelperFtn to be more flexible and support only one path if this affects perf.